examples/daisyworld.jl

```diff
@@ -48,249 +60,269 @@ mutable struct Daisy <: AbstractAgent
     albedo::Float64 # 0-1 fraction
 end
 
+mutable struct Land <: AbstractAgent
+    id::Int
+    pos::Tuple{Int,Int}
+    temperature::Float64
+end
+
+const DaisyWorld = ABM{Union{Daisy, Land}};
+
 # ## World heating
 
 # The surface temperature of the world is heated by its sun, but daisies growing upon it
 # absorb or reflect the starlight -- altering the local temperature.
 
-function suface_temperature!(node::Int, model::ABM{Daisy})
+function update_surface_temperature!(node::Int, model::DaisyWorld)
     ids = get_node_contents(node, model)
-    absorbed_luminosity = if isempty(ids)
+    absorbed_luminosity = if length(ids) == 1 # no daisy
         ## Set luminosity via surface albedo
         (1 - model.surface_albedo) * model.solar_luminosity
     else
         ## Set luminosity via daisy albedo
-        (1 - model[ids[1]].albedo) * model.solar_luminosity
+        (1 - model[ids[2]].albedo) * model.solar_luminosity
     end
-    ## We expect local heating to be 80C for an absorbed luminosity of 1,
+    ## We expect local heating to be 80 ᵒC for an absorbed luminosity of 1,
     ## approximately 30 for 0.5 and approximately -273 for 0.01.
-    local_heating = if absorbed_luminosity > 0
-        72 * log(absorbed_luminosity) + 80
-    else
-        80
-    end
+    local_heating = absorbed_luminosity > 0 ? 72 * log(absorbed_luminosity) + 80 : 80
     ## Surface temperature is the average of the current temperature and local heating.
-    model.temperature[node] = (model.temperature[node] + local_heating) / 2
+    T0 = model[ids[1]].temperature
+    model[ids[1]].temperature = (T0 + local_heating) / 2
 end
 
-function diffuse_temperature!(node::Int, model::ABM{Daisy}; ratio = 0.5)
-    neighbors = node_neighbors(node, model)
-    model.temperature[node] =
-        (1 - ratio) * model.temperature[node] +
-        ## Each neighbor is giving up 1/8 of the diffused
-        ## amount to each of *its* neighbors
-        sum(model.temperature[neighbors]) * 0.125 * ratio
+# In addition, temperature diffuses over time
+function diffuse_temperature!(node::Int, model::DaisyWorld)
+    ratio = get(model.properties, :ratio, 0.5) # diffusion ratio
```

Comment:
Why use this vs a keyword? If a user always constructs their model via the initialise function, they'll not have access to modify this ratio anymore right?

Reply:
If `ratio` is a model parameter, it is used here. What do you mean by "keyword"?

Reply:
Yeah, I follow that. 

But you've come from a function that has `diffuse_temperature!(node::Int, model::ABM{Daisy}; ratio = 0.5)`, to taking a model parameter (fine).

The issue is your `daisyworld` constructor function (beginning line 191) doesn't have ratio as a property. 

The user would therefore need to read the function details to see that they could add such a property, vs the old way where they'd only need to look at the function definition. 

So the question I'm asking is if there is an advantage I'm not seeing to changing this from the keyword (`ratio = 0.5`) method.